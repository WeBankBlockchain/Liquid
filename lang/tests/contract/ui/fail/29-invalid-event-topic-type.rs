use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(event)]
    struct TestEvent {
        i: i32,
        v: Vec<i32>,
        #[liquid(indexed)]
        tv: Vec<i32>,
    }

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) -> () {
            self.env().emit(TestEvent {
                i: 1,
                v: Vec::new(),
                tv: Vec::new(),
            });
        }
    }
}

fn main() {}
