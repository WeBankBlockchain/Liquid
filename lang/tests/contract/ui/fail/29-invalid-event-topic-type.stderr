error[E0277]: the trait bound `std::vec::Vec<i32>: You_Should_Use_An_Valid_Topic_Type` is not satisfied
  --> $DIR/29-invalid-event-topic-type.rs:13:13
   |
13 |         tv: Vec<i32>,
   |             ^^^^^^^^ the trait `You_Should_Use_An_Valid_Topic_Type` is not implemented for `std::vec::Vec<i32>`
